docs/components/codeblock.md

```diff
@@ -0,0 +1,323 @@
+# CodeBlock
+
+**KCodeBlock** is a code block component with built-in search and filter functionalities.
+
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+/>
+```
+
+## Props
+
+### id
+
+* **Type**: `string`
+* **Required**: yes
+
+An ID used to identify form elements like the search input and its label.
+
+### code
+
+* **Type**: `string`
+* **Required**: yes
+
+The code that will be rendered as a text node inside of a `<pre><code></code></pre>` fragment.
+
+### language
+
+* **Type**: `string`
+* **Required**: yes
+
+The syntax language of `props.code` (e.g. `'json'`).
+
+### isSearchable
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Shows an actions bar with a search input and related action buttons.
+
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+```
+
+### isProcessing
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Allows controlling the processing state from outside the component. This allows a parent component to show the processing icon when it’s, for example, currently syntax highlighting the code.
+
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-searchable
+  is-processing
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-processing
+/>
+```
+
+### query
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `''`
+
+Used as the initial value of the search input. Can be used to initialize a code block with a query which was read from client storage.
+
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="vite"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="vite"
+/>
+```
+
+### showCopyButton
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `true`
+
+Controls when to show a copy button which copies the original code (i.e. the value of `props.code`) to the clipboard.
+
+<KCodeBlock
+  id="code-block-show-copy-button"
+  :code="code"
+  language="json"
+  :show-copy-button="false"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-show-copy-button"
+  :code="code"
+  language="json"
+  :show-copy-button="false"
+/>
+```
+
+### showLineNumbers
```

Comment:
**Question** (cc @adamdehaven): What do you think about dropping `props.showLineNumbers` entirely? Perhaps there *are* cases where one wants to display a code block without line numbers, sure, but aren’t those typically cases where a plain `<pre><code></code></pre>` would be sufficient (i.e. they don’t need search)?

I’m asking because this would cut down on the complexity of the code block implementation. Getting hiding line numbers to work well with still showing highlighted line numbers gets tricky and I’m currently resorting to some, well, let’s call it more arcane techniques.

Reply:
I'm fine with line numbers always showing. We can add an enhancement in the future if we decide we need to be able to hide them.